DEVELOPMENT.md

```diff
@@ -429,12 +406,71 @@ git submodule update --init --remote
 git submodule foreach "git fetch --all"
 ```
 
-Then run tests as usual: `cargo test`.
+Then run tests as usual: `cargo test --all-features --all-targets`.
+
+We strive for two kind of tests, the classic unit tests inside of the
+implementation files. Additionally we have a suit of integration tests esp. to
+assert the API the proxy provides from a high-level consumer perspective, these
+can be found under `proxy/tests`. To find out where to place and how to layout
+tests check Rust book [test chapter][rt].
+
+### File structure
+
+We understand the GraphQL API as one potential surface to expose the
```

Comment:
> We understand the GraphQL API as one potential surface to expose the Application domain logic

I am not sure I understand this. Could you explain what you mean? How could we explain this in a simpler way?

Reply:
Maybe:

> The GraphQL API is one potential surface for exposing the application's domain logic.

But I don't understand what "potential" is doing in this sentence. Since we already use it, isn't it "actual"? If so, it could be dropped and the sentence rephrased:

> The GraphQL API exposes the application's domain logic. Therefore we try to treat it as a thin layer...